### ISSUE TITLE
CloudWatch MetricDataQuery Id regex

### ISSUE THREAD

**Issue by dalhaddad** [aws-cloudformation/cfn-lint#2413]
### CloudFormation Lint Version

0.67.0

### What operating system are you using?

Ubuntu

### Describe the bug

When creating an AWS::CloudWatch::Alarm and using metric math, the Id field of each MetricDataQuery may be as short as 1 character long as long as it starts with a lowercase letter, per https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudwatch-alarm-metricdataquery.html. In testing with a metric definition that's only one character long and a letter, the linter complains that

> Id contains invalid characters (Pattern: ^([a-z])([A-Za-z0-9\_]+)$) at Resources/CwInstanceAlarm/Properties/Metrics/0/Id

showing a regex that would require a 2 character minimum length.

### Expected behavior

No linter complaint about a short name like `Id: m`.

### Reproduction template

```yaml
Type: AWS::CloudWatch::Alarm
Properties:
  AlarmDescription: CCI Instance Status Check Failure
  ComparisonOperator: GreaterThanThreshold
  EvaluationPeriods: 1
  Metrics:
    - Id: m
      MetricStat:
        Metric:
          Dimensions:
            - Name: InstanceId
              Value: !Ref Ec2Instance
          MetricName: StatusCheckFailed_Instance
          Namespace: AWS/EC2
        Period: 300
        Stat: Maximum
      ReturnData: False
    - Id: expr
      Expression: IF(DAY(m) == 6 AND HOUR(m) >= 6 AND HOUR(m) < 8, 0, m)
  Threshold: 0
```

**Comment by PatMyron** [comment]
line to change:
https://github.com/aws-cloudformation/cfn-lint/blob/f1c5772bd10966f2ea1088ba96cee786873c3eb9/src/cfnlint/data/ExtendedSpecs/all/03_value_types/aws_cloudwatch.json#L55
